### DISCUSSION TITLE
Why do we need the training labels/annotations if this is a self-supervised training?

### DISCUSSION THREAD

**Discussion by chandlerbing65nm** [open-mmlab/mmselfsup#670] [General] [answered]
I'm confused on why we need to add the training labels/annotations for training.

**Comment by fangyixiao18** [comment, marked as answer]
> I'm confused on why we need to add the training labels/annotations for training.

Can you provide the code scripts? Self-supervised learning doesn't need ground-truth labels, but some methods may use some pseudo labels.

**Reply by chandlerbing65nm** [reply]
> > I'm confused on why we need to add the training labels/annotations for training.
> 
> Can you provide the code scripts? Self-supervised learning doesn't need ground-truth labels, but some methods may use some pseudo labels.

@fangyixiao18 this one:

https://github.com/open-mmlab/mmselfsup/blob/684875d09b68ae654b1d3186190a993274dcc297/configs/selfsup/_base_/datasets/imagenet_simclr.py#L39

Why do I need the training labels for imagenet?

**Reply by fangyixiao18** [reply]
we use ann_file to load image according to the image path in the file, but we don't use labels during training.

**Reply by fangyixiao18** [reply]
https://github.com/open-mmlab/mmselfsup/blob/1.x/docs/en/user_guides/4_pretrain_custom_dataset.md
you can train the model on your dataset, following this document on 1.x branch

**Reply by chandlerbing65nm** [reply]
oh thanks @fangyixiao18. So, if I use my own dataset and format it similar to imagenet, I just need the names of the images listed in a txt file?

**Reply by fangyixiao18** [reply]
Yes, you could try it
